mbuild/compound.py

```diff
@@ -312,8 +323,205 @@ def particles_by_name(self, name):
+            Name(s) of Compound instances to be treated as unique rigid bodies.
+            Compound instances matching this (these) name(s) will be provided 
+            with unique rigid_ids
+        rigid_particles : str or list of str, optional, default=None
+            Name(s) of Compound instances at the bottom of the containment
+            hierarchy (Particles) to be included in rigid bodies. Only Particles
+            matching this (these) name(s) will have their rigid_ids altered to 
+            match the rigid body number.
+
+        Examples
+        --------
+        Creating a rigid benzene
+
+        >>> import mbuild as mb
+        >>> from mbuild.utils.io import get_fn
+        >>> benzene = mb.load(get_fn('benzene.mol2'))
+        >>> benzene.label_rigid_bodies()
+
+        Creating a semi-rigid benzene, where only the carbons are treated as
+        a rigid body
+
+        >>> import mbuild as mb
+        >>> from mbuild.utils.io import get_fn
+        >>> benzene = mb.load(get_fn('benzene.mol2'))
+        >>> benzene.label_rigid_bodies(rigid_particles='C')
+
+        Create a box of rigid benzenes, where each benzene has a unique rigid
+        body ID.
+
+        >>> import mbuild as mb
+        >>> from mbuild.utils.io import get_fn
+        >>> benzene = mb.load(get_fn('benzene.mol2'))
+        >>> benzene.name = 'Benzene'
+        >>> filled = mb.fill_box(benzene,
+        ...                      n_compounds=10,
+        ...                      box=[0, 0, 0, 4, 4, 4])
+        >>> filled.label_rigid_bodies(distinct_bodies='Benzene')
+
+        Create a box of semi-rigid benzenes, where each benzene has a unique
+        rigid body ID and only the carbon portion is treated as rigid.
+
+        >>> import mbuild as mb
+        >>> from mbuild.utils.io import get_fn
+        >>> benzene = mb.load(get_fn('benzene.mol2'))
+        >>> benzene.name = 'Benzene'
+        >>> filled = mb.fill_box(benzene,
+        ...                      n_compounds=10,
+        ...                      box=[0, 0, 0, 4, 4, 4])
+        >>> filled.label_rigid_bodies(distinct_bodies='Benzene',
+        ...                           rigid_particles='C')
+
+        """
+        if discrete_bodies is not None:
+            if isinstance(discrete_bodies, string_types):
+                discrete_bodies = [discrete_bodies]
+        if rigid_particles is not None:
+            if isinstance(rigid_particles, string_types):
+                rigid_particles = [rigid_particles]
+
+        if self.root.max_rigid_id is not None:
+            rigid_id = self.root.max_rigid_id + 1
+            warn("{} rigid bodies already exist.  Incrementing 'rigid_id'"
+                 "starting from {}.".format(rigid_id, rigid_id))
+        else:
+            rigid_id = 0
+
+        for successor in self.successors():
+            if discrete_bodies and successor.name not in discrete_bodies:
+                pass
```

Comment:
If you use `continue` instead of pass here, you can eliminate the need for the `else` branch right after and make for slightly cleaner code.